### PR TITLE
nightshade upgrade

### PR DESCRIPTION
**Pull request by manics**
5.6.0 Nightshade upgrade.

Also adds some molecule tests. The verify step of the webclient test currently hangs on Travis but passes locally, so is disabled. Previously there were no tests at all for these playbooks so this is still an improvement.

### CODE REVIEW

**Comment by snoopycrimecop** [comment]
Empty PR description. Please add a short summary of the PR scope and some testing instructions.

**Comment by snoopycrimecop** [comment]
Empty PR description. Please add a short summary of the PR scope and some testing instructions.

**Comment by snoopycrimecop** [comment]
Empty PR description. Please add a short summary of the PR scope and some testing instructions.

**Comment by snoopycrimecop** [comment]
Empty PR description. Please add a short summary of the PR scope and some testing instructions.

File: nightshade-webclients.yml [hunk 1]
```diff
@@ -114,26 +102,83 @@
...
+        (omero_web_config_set_for_group | default({})),
+        (omero_web_config_set_for_host | default({})))
+      }}
+
+    omero_web_release: "{{ omero_web_release_override | default('5.6.1') }}"
+    omero_figure_release: "{{ omero_figure_release_override | default('4.2.0') }}"
+    omero_fpbioimage_release: "{{ omero_fpbioimage_release_override | default('0.4.0') }}"
+    omero_iviewer_release: "{{ omero_iviewer_release_override | default('0.9.0') }}"
+    omero_parade_release: "{{ omero_parade_release_override | default('0.2.0') }}"
+    # omero_webtagging_autotag_release: "{{ omero_webtagging_autotag_release_override | default('3.0.2') }}"
```
Comment (mtbc) [inline nightshade-webclients.yml]: maybe okay for 3.1.0 now

File: ome-dundeeomero.yml [hunk 2]
```diff
@@ -245,29 +256,38 @@
     #omero_server_db_dumpdir_parent: /tmp
     #omero_server_db_dumpdir_name: nightly-pg_dump_omero.dir
 
-    ice_version: "3.6"
+    nginx_version: 1.16.1
```
Comment (mtbc) [inline ome-dundeeomero.yml]: Should I maybe bump demo up to this from its current 1.14.2?

File: nightshade-webclients.yml (hunk 1 above)
Comment (manics) [reply nightshade-webclients.yml]: Tat will require a another deployment run, leave for the next PR?

File: ome-dundeeomero.yml (hunk 2 above)
Comment (manics) [reply ome-dundeeomero.yml]: There's no harm in updating demo, but I don't think there's a strong need to at the moment either.

File: nightshade-webclients.yml (hunk 1 above)
Comment (joshmoore) [reply nightshade-webclients.yml]: web tagging access.log hits for reference:
```
     80 2019-Jan
     95 2019-Feb
    127 2019-Mar
     81 2019-Apr
    169 2019-May
    499 2019-Jun
    207 2019-Jul
    164 2019-Aug
   1554 2019-Sep
    217 2019-Oct
    440 2019-Nov
     89 2019-Dec
     67 2020-Jan
```

File: ome-dundeeomero.yml [hunk 3]
```diff
@@ -245,29 +256,38 @@
...
     postgresql_version: "9.6"
     filesystem: "xfs"
-    omero_figure_release: "{{ omero_figure_release_override | default('v4.0.2') }}"
-    omero_server_config_set:
+    omero_figure_release: "{{ omero_figure_release_override | default('4.2.0') }}"
+
+    omero_web_config_set_production:
```
Comment (joshmoore) [inline ome-dundeeomero.yml]: This surprises me for "omero.db.poolsize"

File: ome-dundeeomero.yml [hunk 4]
```diff
@@ -245,29 +256,38 @@
...
       omero.throttling.method_time.error: 60000
-      omero.Ice.Default.Host: "{{ omero_server_ice_default_host }}"
-      Ice.Admin.Endpoints: "{{ omero_server_ice_admin_endpoints }}"
-      omero.data.dir: "{{ omero_server_datadir  }}"
+      omero.Ice.Default.Host: "{{ omero_server_ice_default_host | default('127.0.0.1') }}"
+      Ice.Admin.Endpoints: "{{ omero_server_ice_admin_endpoints | default('tcp -h 127.0.0.1') }}"
+      omero.data.dir: "{{ omero_server_datadir | default('/OMERO') }}"
+
+    # Production config can't be tested in molecule
+    omero_server_config_set: "{{ molecule_test | default(False) | ternary({}, omero_web_config_set_production) }}"
```
Comment (joshmoore) [inline ome-dundeeomero.yml]: The propagation of `molecule_test` is a bit worrisome.

File: templates/nginx-omero.conf.j2 [hunk 5]
```diff
@@ -39,9 +39,9 @@ server {
     location / {
         error_page 502 @maintenance;
         # return 502;
-        # checks for static file, if not found proxy to {{ nginx_omero_web_server_address }}
+        # checks for static file, if not found proxy to nginx_omero_web_server_address
     
-        proxy_pass              {{ nginx_omero_web_server_address }};
+        proxy_pass              {{ nginx_omero_web_server_address | default('https://www.openmicroscopy.org/') }};
```
Comment (joshmoore) [inline templates/nginx-omero.conf.j2]: This doesn't seem appropriate.

File: ome-dundeeomero.yml (hunk 4 above)
Comment (manics) [reply ome-dundeeomero.yml]: I tried using tags without success. If you get it working let me know!

File: templates/nginx-omero.conf.j2 (hunk 5 above)
Comment (manics) [reply templates/nginx-omero.conf.j2]: I need a working URL, otherwise Nginx won't start in the molecule tests

File: ome-dundeeomero.yml (hunk 3 above)
Comment (manics) [reply ome-dundeeomero.yml]: It's what's currently there, can you open a new PR after this is merged with whatever you think is appropriate?

**Comment by manics** [comment]
webtagging: how about next week? Or if you want to adjust `omero.db.poolsize` then whenever we decide to make that change.

File: ome-dundeeomero.yml (hunk 3 above)
Comment (joshmoore) [reply ome-dundeeomero.yml]: It's less the value and more the fact that it's set on `omero_web_*` at all. Why did this change from `omero_server` to `omero_web`?

**Comment by joshmoore** [comment]
> webtagging: how about next week? 

Works for me, but it'd be good to get nightshade done & dusted.

File: ome-dundeeomero.yml (hunk 3 above)
Comment (manics) [reply ome-dundeeomero.yml]: Sorry, it's a typo. Fixed.

Comment (joshmoore) [reply ome-dundeeomero.yml]: :+1: does the playbook now show a diff, because if so, we may need to re-deploy ASAP.

Comment (manics) [reply ome-dundeeomero.yml]: It shouldn't, the var is only used indirectly via it's reference

**Comment by manics** [comment]
Last two commits add `OMERO_TMPDIR` and webtagging. If everyone's happy with these settings downtime can be scheduled.

File: ome-dundeeomero.yml [hunk 6]
```diff
@@ -245,29 +265,43 @@
...
+    # Production config can't be tested in molecule
+    omero_server_config_set: "{{ molecule_test | default(False) | ternary({}, omero_server_config_set_production) }}"
+    omero_server_python_addons:
+      # For OMERO.figure script
+      - reportlab
+      - markdown
+
+    # Workaround lack of restriction on temp file sizes
+    # https://github.com/ome/omero-web/issues/118
+    # The downside is that it won't be automatically cleared out
```
Comment (joshmoore) [inline ome-dundeeomero.yml]: The workaround for the downside would be to move `/tmp` to its own partition?

**Comment by manics** [comment]
The last two commits aren't deployed as it requires a restart and therefore downtime. I'll send out an announcement for Monday morning.

**Comment by manics** [comment]
Last few commits deployed this morning

**Pull request merged**